resolver.go

```diff
@@ -344,172 +336,55 @@ func (r *Resolver) LookupAnswersForType(name string, rrType uint16) (answers []d
-        labels, ok := dns.IsDomainName(node.Value)
-
-        if (ok && labels > 0) {
-            rr = &dns.PTR{header, dns.Fqdn(node.Value)}
-        } else {
-            err = &NodeConversionError{
-                Node: node,
-                Message: fmt.Sprintf("Value '%s' isn't a valid domain name", node.Value),
-                AttemptedType: dns.TypePTR}
-        }
-        return
-    },
-
-    dns.TypeSRV: func (node *etcd.Node, header dns.RR_Header) (rr dns.RR, err error) {
-        parts := strings.SplitN(node.Value, "\t", 4)
-
-        if len(parts) != 4 {
-            err = &NodeConversionError{
-                Node: node,
-                Message: fmt.Sprintf("Value %s isn't valid for SRV", node.Value),
-                AttemptedType: dns.TypeSRV}
-        } else {
-
-            priority, err := strconv.ParseUint(parts[0], 10, 16)
-            if err != nil {
-                return nil, err
-            }
-
-            weight, err := strconv.ParseUint(parts[1], 10, 16)
-            if err != nil {
-                return nil, err
-            }
-
-            port, err := strconv.ParseUint(parts[2], 10, 16)
-            if err != nil {
-                return nil, err
-            }
-
-            target := dns.Fqdn(parts[3])
-
-            rr = &dns.SRV{
-                header,
-                uint16(priority),
-                uint16(weight),
-                uint16(port),
-                target}
-        }
-        return
-    },
-
-    dns.TypeSOA: func (node *etcd.Node, header dns.RR_Header) (rr dns.RR, err error) {
-        parts := strings.SplitN(node.Value, "\t", 6)
-
-        if len(parts) < 6 {
-            err = &NodeConversionError{
-                Node: node,
-                Message: fmt.Sprintf("Value %s isn't valid for SOA", node.Value),
-                AttemptedType: dns.TypeSOA}
-        } else {
-            refresh, err := strconv.ParseUint(parts[2], 10, 32)
-            if err != nil {
-                return nil, err
-            }
-
-            retry, err := strconv.ParseUint(parts[3], 10, 32)
-            if err != nil {
-                return nil, err
-            }
+// RRSetExists returns true if RRs exist for the given name and type (value independent)
+func (r *Resolver) RRSetExists(name string, rrType uint16) (exists bool, err error) {
+    answers, err := r.LookupAnswersForType(dns.Fqdn(name), rrType)
+    if err != nil {
+        return false, err
+    }
 
-            expire, err := strconv.ParseUint(parts[4], 10, 32)
-            if err != nil {
-                return nil, err
-            }
+    return len(answers) > 0, nil
+}
 
-            minttl, err := strconv.ParseUint(parts[5], 10, 32)
-            if err != nil {
-                return nil, err
+// RRSetMatches checks that the set of records in the DNS for the given name
+// and type *exactly* match the given RRs: their data must match, and there must
+// be no more or less RRs
```

Comment:
I think this might be incorrect – I think i misread the RFC Having some trouble checking existing implementations (man, bind code is hard to read :-/ ) but I think the "RRSet Exists (Value-Dependent" prereq means that _some set_ of exactly-matching RRs must exist, but permits other records to also exist. I think I might have un-done this in my refactor and @tarnfeld had it right to begin with.

References: [Net::DNS](https://www.net-dns.org/docs/Net/DNS.html), [RFC (section 2.4.2)](https://tools.ietf.org/html/rfc2136#page-7)